lib/processModelMaterialsCommon.js

```diff
@@ -208,13 +208,39 @@ function generateTechnique(gltf, khrMaterialsCommon, lightParameters, optimizeFo
     var fragmentShaderId = getNextId(shaders, 'fragmentShader', fragmentShaderCount);
     var programId = getNextId(programs, 'program', programCount);
 
-    var hasNormals = (lightingModel !== 'CONSTANT');
+
+    // Check if normals exist
+    var meshesHaveNormals = false;
+    var meshes = gltf.meshes;
+    for (var meshId in meshes) {
+        if (meshes.hasOwnProperty(meshId)) {
+            var mesh = meshes[meshId];
+            if (mesh.hasOwnProperty('primitives')) {
+                for (var p = 0; p < mesh.primitives.length; p++) {
+                    if (mesh.primitives[p].material === materialId) {
+                        if (defined(mesh.primitives[p].attributes.NORMAL)) {
+                            meshesHaveNormals = true;
+                            break;
```

Comment:
Might be worth putting this whole section into a separate function and using `return` instead to break out of both loops.